lib/i18n/js/segment.rb

```diff
@@ -14,12 +16,21 @@ def initialize(file, translations, options = {})
 
       # Saves JSON file containing translations
       def save!
-        FileUtils.mkdir_p File.dirname(self.file)
+        if file =~ LOCALE_INTERPOLATOR
+          I18n.available_locales.each do |locale|
+            write_file(file_for_locale(locale), self.translations.slice(locale))
```

Comment:
Either always use `self.something`, or don't use `self` all the time (except using setter)
code with mixed style is hard to read